### PR TITLE
Hide blank fields on models

### LINKED ISSUE #11: Don't show blank fields on models
If a user, project, or challenge don't have social links, creators, etc. we shouldn't display the heading.

Required for: #10

### PR DESCRIPTION
**Pull request by dulldesk**
A heading of a field (e.g. links, tags) are not displayed if the field is empty. 

Fixes #11

### CODE REVIEW

File: app/templates/userhome.html [hunk 1]
```diff
@@ -61,8 +66,19 @@ <h2 class="header-margin">Profile</h2>
...
+    {# TODO: Replace hardcoding in the line below #}
+    {% if not profile.description and not profile.badges.all and not links.all and not profile.tags.all %}
+    <div class="col-12">
+      {% if profile != request.user.profile %}
+      <p>{{ profile.username }} has not given any user details.</p>
+      {% else %}
+      <p>No user details! Go to <a href="{% url 'edit_profile' %}">Edit Profile</a> to add something about yourself.</p>
+      {% endif %}
+    </div>
+    {% else %}
```
Comment (dulldesk) [inline app/templates/userhome.html]: I had hardcoded the if statement on line 70, it should be replaced.

File: app/templates/initiative.html [hunk 2]
```diff
@@ -104,6 +106,7 @@ <h2>{{ initiative.name }}</h2>
         {% endfor %}
 
         <!-- links -->
+        {% if links.all %}
         <div class="tile level r">
```
Comment (fluix-dev) [inline app/templates/initiative.html]: Indent `{% ... %}` blocks -- this applies to many other places.

File: app/templates/initiative.html [hunk 3]
```diff
@@ -39,6 +39,7 @@ <h2>{{ initiative.name }}</h2>
         {% endif %}
 
         <!-- tags -->
+        {% if initiative.tags.all %}
```
Comment (fluix-dev) [inline app/templates/initiative.html]: Let's use `.exists` instead, same for all the other places where it's a relational field.

**Review by fluix-dev — CHANGES_REQUESTED** [review]
Otherwise looks good, thanks!

File: app/templates/challenge.html [hunk 4]
```diff
@@ -1,44 +1,44 @@
...
-        {% for tag in project.tags.all|slice:':3' %}
-        <div class="tag white" style="background-color: {{ tag.color }}">{{ tag.name }}</div>
-        {% endfor %}
-      </div>
-    </div>
+    {% endfor %}
   </div>
-  {% endfor %}
-</div>
 {% endblock %}
```
Comment (fluix-dev) [inline app/templates/challenge.html]: Add newline.

Comment (fluix-dev) [reply app/templates/challenge.html]: There's a special symbol [here](https://github.com/KuriusMTL/stratus/pull/21/files#diff-d6d534418b232792a9f4b23c888def36R44).

File: app/templates/userhome.html (hunk 1 above)
Comment (fluix-dev) [reply app/templates/userhome.html]: I think it's alright tbh, so remove the `TODO`.

**Pull request merged**